src/jit/emitxarch.cpp

```diff
@@ -534,7 +536,10 @@ unsigned emitter::emitOutputRexOrVexPrefixIfNeeded(instruction ins, BYTE* dst, s
         emitOutputByte(dst + 2, vexPrefix & 0xFF);
         return 3;
     }
-    else if (code > 0x00FFFFFFFFLL)
+#endif // FEATURE_AVX_SUPPORT
+
+#ifdef _TARGET_AMD64_
```

Comment:
The REX prefixes are only used for AMD64. When set in the AVX prefix, they are only used for extended register usage.